lib/pbench/server/database/models/template.py

```diff
@@ -165,6 +165,31 @@ def __str__(self) -> str:
         """
         return f"{self.name}: {self.index_template}"
 
+    def _decode(self, exception: IntegrityError) -> Exception:
+        """
+        Decode a SQLAlchemy IntegrityError to look for a recognizable UNIQUE
+        or NOT NULL constraint violation. Return the original exception if
+        it doesn't match.
+
+        Args:
+            exception: An IntegrityError to decode
+
+        Returns:
+            a more specific exception, or the original if decoding fails
+        """
+        # Postgres engine returns (code, message) but sqlite3 engine only
+        # returns (message); so always take the last element.
+        cause = exception.orig.args[-1]
+        Database.db_session.rollback()
```

Comment:
This line is duplicated in the callers, so it can/should be removed from here.

Reply:
Well, dang. And now this really is going to wait until tomorrow, because when I changed that I lost a bunch of unit tests in ways that make absolutely no sense at all. It must be the rebase, always I was sure I'd run it before the last commit; but I'm not dealing with that now.